core/trino-spi/src/test/java/io/trino/spi/predicate/TestTupleDomain.java

```diff
@@ -647,14 +649,16 @@ public void testTransformFailsWithNonUniqueMapping()
 
         TupleDomain<Integer> domain = TupleDomain.withColumnDomains(domains);
 
-        domain.transform(input -> "x");
+        assertThatThrownBy(() -> domain.transform(input -> "x")).isInstanceOf(IllegalArgumentException.class)
+                .hasMessage(format("Every argument must have a unique mapping. %s maps to %s and %s",
+                        2, Domain.singleValue(BIGINT, 2L), Domain.singleValue(BIGINT, 1L)));
```

Comment:
Let's use a constant string as the expected message, this will help us ensure the message is not gabled and remains human-readable

(the easiest way to do this is have `.hasMessage("xxx")`, run the test and copy whatever message shows up as the expected)